### PR TITLE
Altair Sync committee store service

### PR DESCRIPTION
**Pull request by terencechain**
Part of #8982 

Follow up on #8982

Add service to wrap around sync committee store ~~add pruning for caches in sync committee store~~

### CODE REVIEW

File: beacon-chain/operations/synccommittee/service.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+// Stop the sync committee store service.
+func (s *Service) Stop() error {
+	defer s.cancel()
+	return nil
+}
+
+// Status returns the current service err if there's any.
+func (s *Service) Status() error {
```
Comment (prestonvanloon) [inline beacon-chain/operations/synccommittee/service.go:39]: Seems like this thing doesn't do anything if the genesis time is not set. Want to return an error if genesis time is not set?

File: beacon-chain/operations/synccommittee/prune.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+// This prunes expired sync committee signatures from the store.
+func (s *Service) pruneExpiredSyncCommitteeSignatures() {
+	currentSlot := helpers.CurrentSlot(s.genesisTime)
+	s.store.signatureLock.Lock()
+	defer s.store.signatureLock.Unlock()
+
+	// Delete the sync committee signatures from 2 slots back.
+	// Doesn't matter when in current slot the deletion happen,
+	// and this is the simplest and safest approach.
+	expiredSlot := currentSlot - 2
```
Comment (prestonvanloon) [inline beacon-chain/operations/synccommittee/prune.go:41]: Please handle the case when current slot is < 2

File: beacon-chain/operations/synccommittee/prune.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+func (s *Service) pruneExpiredSyncCommitteeContributions() {
+	currentSlot := helpers.CurrentSlot(s.genesisTime)
+
+	s.store.contributionLock.Lock()
+	defer s.store.contributionLock.Unlock()
+
+	// Delete the sync committee signatures from 2 slots back.
+	// Doesn't matter when in current slot the deletion happen,
+	// and this is the simplest and safest approach.
+	expiredSlot := currentSlot - 2
```
Comment (prestonvanloon) [inline beacon-chain/operations/synccommittee/prune.go:55]: Same here

File: beacon-chain/operations/synccommittee/service.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/operations/synccommittee/service.go:39]: Makes sense to me

File: beacon-chain/operations/synccommittee/prune.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+// This prunes expired sync committee signatures from the store.
+func (s *Service) pruneExpiredSyncCommitteeSignatures() {
+	currentSlot := helpers.CurrentSlot(s.genesisTime)
+	s.store.signatureLock.Lock()
+	defer s.store.signatureLock.Unlock()
+
+	// Delete the sync committee signatures from 2 slots back.
+	// Doesn't matter when in current slot the deletion happen,
+	// and this is the simplest and safest approach.
```
Comment (prestonvanloon) [inline beacon-chain/operations/synccommittee/prune.go:41]: This sentence doesn't make sense. Can you clarify?

**Review by prestonvanloon — CHANGES_REQUESTED** [review]
Mentioned some comments on discord. I feel like this is a lot of complexity for a cache with 2 keys in it. Why not use some FIFO? Or even some primitive data structure to hold the last 2 slots in cache, when a third one is added then evict the oldest one (a very basic FIFO).

File: beacon-chain/operations/synccommittee/prune.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	expiredSlot := types.Slot(0)
+	// Prevent underflow when slot is lesser than two.
+	if currentSlot > 2 {
+		expiredSlot = currentSlot - 2
+	}
+	delete(s.store.signatureCache, expiredSlot)
```
[the remark this answers is not in the document]
Comment (terencechain) [reply beacon-chain/operations/synccommittee/prune.go]: You are right, We should just return early

**Comment by terencechain** [comment]
> Mentioned some comments on discord. I feel like this is a lot of complexity for a cache with 2 keys in it. Why not use some FIFO? Or even some primitive data structure to hold the last 2 slots in cache, when a third one is added then evict the oldest one (a very basic FIFO).

After more thinking, I do think an ordered fifo is a better approach. Let's take out the pruning part and keep the service part for merging

**Comment by terencechain** [comment]
Re-visit later when needed

**Pull request closed without merging**